examples/cu_consolemon/src/sysinfo.rs

```diff
@@ -0,0 +1,220 @@
+};
+
+use pfetch_logo_parser::{Color, Logo, LogoPart};
+use std::{env, fmt::Display, str::FromStr};
+
+#[derive(Debug, PartialEq)]
+pub enum PfetchInfo {
+    Ascii,
+    Title,
+    Os,
+    Host,
+    Kernel,
+    Uptime,
+    Cpu,
+    Memory,
+    Shell,
+    Editor,
+    Wm,
+    De,
+    Palette,
+    BlankLine,
+}
+
+impl Display for PfetchInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", format!("{self:?}").to_lowercase())
+    }
+}
+
+impl FromStr for PfetchInfo {
+    type Err = String;
+
+    fn from_str(info: &str) -> Result<Self, Self::Err> {
+        match info {
+            "ascii" => Ok(PfetchInfo::Ascii),
+            "title" => Ok(PfetchInfo::Title),
+            "os" => Ok(PfetchInfo::Os),
+            "host" => Ok(PfetchInfo::Host),
+            "kernel" => Ok(PfetchInfo::Kernel),
+            "uptime" => Ok(PfetchInfo::Uptime),
+            "cpu" => Ok(PfetchInfo::Cpu),
+            "memory" => Ok(PfetchInfo::Memory),
+            "shell" => Ok(PfetchInfo::Shell),
+            "editor" => Ok(PfetchInfo::Editor),
+            "wm" => Ok(PfetchInfo::Wm),
+            "de" => Ok(PfetchInfo::De),
+            "palette" => Ok(PfetchInfo::Palette),
+            unknown_info => Err(format!("Unknown pfetch info: {unknown_info}")),
+        }
+    }
+}
+
+// Struct for the various readouts
+pub struct Readouts {
+    pub general_readout: GeneralReadout,
+    pub package_readout: PackageReadout,
+    pub memory_readout: MemoryReadout,
+    pub kernel_readout: KernelReadout,
+}
+
+pub fn get_info(info: &PfetchInfo, readouts: &Readouts) -> Option<String> {
+    match info {
+        PfetchInfo::Ascii => None,
+        PfetchInfo::Title => pfetch::user_at_hostname(
+            &readouts.general_readout,
+            &dotenvy::var("USER").ok(),
+            &dotenvy::var("HOSTNAME").ok(),
+        ),
+        PfetchInfo::Os => pfetch::os(&readouts.general_readout),
+        PfetchInfo::Host => pfetch::host(&readouts.general_readout),
+        PfetchInfo::Kernel => pfetch::kernel(&readouts.kernel_readout),
+        PfetchInfo::Uptime => pfetch::uptime(&readouts.general_readout),
+        PfetchInfo::Cpu => pfetch::cpu(&readouts.general_readout),
+        PfetchInfo::Memory => pfetch::memory(&readouts.memory_readout),
+        PfetchInfo::Shell => pfetch::shell(&readouts.general_readout),
+        PfetchInfo::Editor => Some(env::var("EDITOR").unwrap_or_else(|_| "nvim".to_string())),
+        PfetchInfo::Wm => pfetch::wm(&readouts.general_readout),
+        PfetchInfo::De => pfetch::de(&readouts.general_readout),
+        PfetchInfo::Palette => Some("Color Palette".to_string()), // Simplified palette display
+        PfetchInfo::BlankLine => Some("".to_string()),
+    }
+}
+
+// Function to render the gathered info along with ASCII logo
+pub fn pfetch_info() -> String {
+    let readouts = Readouts {
```

Comment:
Nit: Probably worth just making a `::default()` method on `Readouts` so you can just call `Readouts::default()` wherever you want the default initialization.

Reply:
I needed to copy paste some entry point from this library using libmacchina and now I regret it lol, let me fix this thanks.